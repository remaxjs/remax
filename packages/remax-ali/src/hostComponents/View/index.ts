import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

export interface ViewProps {
  readonly dataset?: DOMStringMap;
  id?: string;
  className?: string;
  style?: React.CSSProperties;
  disableScroll?: boolean;
  hoverClassName?: string;
  hoverStartTime?: number;
  hoverStayTime?: number;
  hidden?: boolean;
  animation?: any;
  hoverStopPropagation?: boolean;
  slot?: string;
  onTap?: (e: any) => void;
  catchTap?: (e: any) => void;
  onClick?: (e: any) => void;
  catchClick?: (e: any) => void;
  onTouchStart?: (e: any) => void;
  onTouchMove?: (e: any) => void;
  onTouchEnd?: (e: any) => void;
  onTouchCancel?: (e: any) => void;
  onLongTap?: (e: any) => void;
  onTransitionEnd?: (e: any) => void;
  onAnimationIteration?: (e: any) => void;
  onAnimationStart?: (e: any) => void;
  onAnimationEnd?: (e: any) => void;
  onAppear?: (e: any) => void;
  onDisappear?: (e: any) => void;
  onFirstAppear?: (e: any) => void;
}

export const View = createHostComponent<ViewProps>('view') as React.ComponentType<ViewProps>;
