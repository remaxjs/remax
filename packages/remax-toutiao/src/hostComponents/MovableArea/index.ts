import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

import { BaseProps } from '../../types/component';

/**
 * https://microapp.bytedance.com/docs/zh-CN/mini-app/develop/component/view-container/movable-area
 * 基础库 2.25.0 开始支持本组件。
 */
export interface MovableAreaProps extends BaseProps {
  // 当里面的 movable-view 设置为支持双指缩放时，设置此值可将缩放手势生效区域修改为整个 movable-area,默认:false
  scaleArea?: boolean;
}

export const MovableArea: React.ComponentType<MovableAreaProps> = createHostComponent<MovableAreaProps>('movable-area');

MovableArea.defaultProps = {
  scaleArea: false,
};
