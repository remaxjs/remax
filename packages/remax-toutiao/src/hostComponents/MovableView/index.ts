import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps, GenericEvent } from '../../types/component';

/***
 * https://microapp.bytedance.com/docs/zh-CN/mini-app/develop/component/view-container/movable-view
 * 基础库 2.25.0 开始支持本组件。
 * 可移动的视图容器，在页面中可以拖拽滑动。movable-view 必须在 movable-area 组件中，并且必须是直接子节点，否则不能移动。
 */
export interface MovableViewProps extends BaseProps {
  // id属性
  id?: string;
  // 类名属性
  className?: string;
  // 样式属性
  style?: React.CSSProperties;
  // 宽度 默认为10px
  width?: number;
  // 高度 默认为10px
  height?: number;
  // movable-view 的移动方向，属性值有 all、vertical、horizontal、默认:none
  direction?: string;
  // movable-view 是否带有惯性，默认:false
  inertia?: boolean;
  // 超过可移动区域后，movable-view 是否还可以移动，默认:false
  outOfBounds?: boolean;
  // 定义 x 轴方向的偏移，如果 x 的值不在可移动范围内，会自动移动到可移动范围；改变 x 的值会触发动画
  x?: number | string;
  // 定义 y 轴方向的偏移，如果 y 的值不在可移动范围内，会自动移动到可移动范围；改变 y 的值会触发动画
  y?: number | string;
  // 阻尼系数，用于控制 x 或 y 改变时的动画和过界回弹的动画，值越大移动越快，默认:20
  damping?: number;
  // 摩擦系数，用于控制惯性滑动的动画，值越大摩擦力越大，滑动越快停止；必须大于 0，否则会被设置成默认:2
  friction?: number;
  // 是否禁用，默认为否
  disabled?: boolean;
  // 是否支持双指缩放，默认缩放手势生效区域是在 movable-view 内，默认否
  scale?: boolean;
  // 定义缩放倍数最小值，，默认:0.5
  scaleMin?: number;
  // 定义缩放倍数最大值，，默认:10
  scaleMax?: number;
  // 定义缩放倍数，取值范围为 0.5 - 10,默认:1
  scaleValue?: number;
  // 是否使用动画，默认:true
  animation?: boolean;
  // 拖动过程中触发的事件，event.detail = {x, y, source}
  onTouchStart?: (e: TouchEvent) => void;
  // 拖动过程中触发的事件，event.detail = {x, y, source}
  onTouchMove?: (e: TouchEvent) => void;
  // 拖动过程中触发的事件，event.detail = {x, y, source}
  onTouchEnd?: (e: TouchEvent) => void;
  // 拖动过程中触发的事件，event.detail = {x, y, source}
  onTouchCancel?: (e: TouchEvent) => void;
  // 拖动过程中触发的事件，event.detail = {x, y, source}
  onTouchChange?: (e: TouchEvent) => void;
  // 拖动过程中触发的事件，event.detail = {x, y, source}
  onChange?: (event: GenericEvent) => any;
  // 缩放过程中触发的事件，event.detail = {x, y, scale}
  onScale?: (event: GenericEvent) => any;
  // 初次手指触摸后移动为横向的移动时触发，如果 catch 此事件，则意味着 touchmove 事件也被 catch
  hTouchmove?: (event: GenericEvent) => any;
  // 初次手指触摸后移动为纵向的移动时触发，如果 catch 此事件，则意味着 touchmove 事件也被 catch
  vTouchmove?: (event: GenericEvent) => any;
}

export const MovableView: React.ComponentType<MovableViewProps> = createHostComponent<MovableViewProps>('movable-view');

MovableView.defaultProps = {
  width: 10,
  height: 10,
  direction: 'none',
  inertia: false,
  outOfBounds: false,
  damping: 20,
  friction: 2,
  disabled: false,
  scale: false,
  scaleMin: 0,
  scaleMax: 10,
  scaleValue: 1,
  animation: true,
};
