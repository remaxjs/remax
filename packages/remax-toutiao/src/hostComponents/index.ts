export * from './View';
export * from './Input';
export * from './Textarea';
export * from './Video';
export * from './Swiper';
export * from './ScrollView';
export * from './SwiperItem';
export * from './Icon';
export * from './Text';
export * from './RichText';
export * from './Progress';
export * from './Button';
export * from './CheckboxGroup';
export * from './Checkbox';
export * from './Form';
export * from './Label';
export * from './Picker';
export * from './PickerView';
export * from './PickerViewColumn';
export * from './RadioGroup';
export * from './Radio';
export * from './Slider';
export * from './Switch';
export * from './Navigator';
export * from './Image';
export * from './Canvas';
export * from './WebView';
export * from './MovableArea';
export * from './MovableView';
