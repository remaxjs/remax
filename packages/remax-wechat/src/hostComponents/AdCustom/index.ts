import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps, GenericEvent } from '../../types/component';

export interface AdCustomProps extends BaseProps {
  /** 广告单元id，可在小程序管理后台的流量主模块新建 2.10.4; */
  unitId: string;
  /** 广告自动刷新的间隔时间，单位为秒，参数值必须大于等于30（该参数不传入时 Banner 广告不会自动刷新） 2.10.4 */
  AdIntervals?: number;
  /** 广告加载成功的回调 2.10.4 */
  onLoad?: (event: GenericEvent) => any;
  /** 广告加载失败的回调，event.detail = {errCode: 1002} 2.10.4 */
  onError?: (event: GenericEvent) => any;
}
/**
 * 原生模版广告
 * @see https://developers.weixin.qq.com/miniprogram/dev/component/ad-custom.html
 */
export const AdCustom: React.ComponentType<AdCustomProps> = createHostComponent<AdCustomProps>('ad-custom');
