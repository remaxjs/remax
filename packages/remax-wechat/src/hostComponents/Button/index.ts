import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps, GenericEvent } from '../../types/component';

export interface ButtonProps extends BaseProps {
  /**
   * 1.0.0
   * default	默认大小
   * mini	小尺寸
   */
  size?: 'default' | 'mini';
  /**
   * 1.0.0
   * primary	绿色
   * default	白色
   * warn	红色
   */
  type?: 'primary' | 'default' | 'warn';

  /**
   * 1.0.0
   * 按钮是否镂空，背景色透明
   */
  plain?: boolean;
  /**
   * 1.0.0
   * 是否禁用
   */
  disabled?: boolean;
  /**
   * 1.0.0
   * 名称前是否带 loading 图标
   */
  loading?: boolean;
  /**
   * 1.0.0
   * submit	提交表单
   * reset	重置表单
   */
  formType?: 'submit' | 'reset';
  /**
   * - 微信小程序 & QQ 小程序
   * contact	打开客服会话，如果用户在会话中点击消息卡片后返回小程序，可以从 bindcontact 回调中获得具体信息	1.1.0
   * share	触发用户转发，使用前建议先阅读使用指引	1.2.0
   * getPhoneNumber	获取用户手机号，可以从 bindgetphonenumber 回调中获取到用户信息	1.2.0
   * getUserInfo	获取用户信息，可以从 bindgetuserinfo 回调中获取到用户信息	1.3.0
   * launchApp	打开 App，可以通过 app-parameter 属性设定向 App 传的参数	1.9.5
   * openSetting	打开授权设置页	2.0.7
   * feedback	打开“意见反馈”页面，用户可提交反馈内容并上传日志，开发者可以登录小程序管理后台后进入左侧菜单“客服反馈”页面获取到反馈内容	2.1.0
   *
   * - QQ 小程序
   * openGroupProfile	呼起群资料卡页面，可以通过 group-id 属性设定需要打开的群资料卡的群号，同时 app.json 中必须配置 groupIdList（数量不超过 10 个），表明可以打开群资料卡的群号	1.4.7
   * addFriend	添加好友，对方需要通过该小程序进行授权，允许被加好友后才能调用成功	1.0.0
   * addColorSign	添加彩签，点击后添加状态有用户提示，无回调	1.10.0
   * openPublicProfile	打开公众号资料卡，可以通过 public-id 属性设定需要打开的公众号资料卡的号码，同时 app.json 中必须配置 publicIdList（目前只能配置 1 个），表明可以打开的公众号资料卡的号码	1.12.0
   * addGroupApp	添加群应用（只有管理员或群主有权操作，建议先调用 qq.getGroupInfo 获取当前用户是否为管理员，如果是管理员才显示该按钮），添加后给 button 绑定 bindaddgroupapp 事件接收回调数据	1.16.0
   * shareMessageToFriend	在自定义开放数据域组件中,向指定好友发起分享	1.17.0
   * addToFavorites	收藏当前页面，点击按钮后会触发 Page.onAddToFavorites 方法	1.19.0
   */
  openType?:
    | 'contact'
    | 'share'
    | 'getPhoneNumber'
    | 'getUserInfo'
    | 'launchApp'
    | 'openSetting'
    | 'feedback'
    | 'openGroupProfile'
    | 'addFriend'
    | 'addColorSign'
    | 'openPublicProfile'
    | 'addGroupApp'
    | 'shareMessageToFriend'
    | 'addToFavorites';
  /**
   * 1.0.0
   * 指定按钮按下去的样式类。当 hover-class="none" 时，没有点击态效果
   */
  hoverClassName?: string;
  /**
   * 1.5.0
   * 指定是否阻止本节点的祖先节点出现点击态
   */
  hoverStopPropagation?: boolean;
  /**
   * 1.0.0
   * 按住后多久出现点击态，单位毫秒
   */
  hoverStartTime?: number;
  /**
   * 1.0.0
   * 手指松开后点击态保留时间，单位毫秒
   */
  hoverStayTime?: number;
  /**
   * 1.3.0
   * en	英文
   * zh_CN	简体中文
   * zh_TW	繁体中文
   */
  lang?: 'en' | 'zh_CN' | 'zh_TW';
  /**
   * 1.4.0
   * 会话来源，open-type="contact"时有效
   */
  sessionFrom?: string;
  /**
   * 1.5.0
   * 会话内消息卡片标题，open-type="contact"时有效
   */
  sendMessageTitle?: string;
  /**
   * 1.5.0
   * 会话内消息卡片点击跳转小程序路径，open-type="contact"时有效
   */
  sendMessagePath?: string;
  /**
   * 1.5.0
   * 会话内消息卡片图片，open-type="contact"时有效
   */
  sendMessageImg?: string;
  /**
   * 1.9.5
   * 打开 APP 时，向 APP 传递的参数，open-type=launchApp时有效
   */
  appParameter?: string;
  /**
   * QQ 小程序 1.0.0
   * 打开 APP 时，应用的包名（Android）
   */
  appPackageName?: string;
  /**
   * QQ 小程序 1.0.0
   * 打开 APP 时，应用的 Bundle ID（iOS）
   */
  appBundleID?: string;
  /**
   * QQ 小程序 1.0.0
   * 打开 APP 时，应用在 QQ 互联中的 AppID
   */
  appConnectID?: string;
  /**
   * QQ 小程序 1.0.0
   * 添加好友时，对方的 openid
   */
  openId?: string;
  /**
   * QQ 小程序 1.0.0
   * 打开群资料卡时，传递的群号
   */
  groupId?: string;
  /**
   * QQ 小程序 1.12.0
   * 打开公众号资料卡时，传递的号码
   */
  publicId?: string;
  /**
   * QQ 小程序 1.4.4
   * 分享类型集合，标志位形式
   */
  shareType?: number;
  /**
   * QQ 小程序 1.15.0
   * 分享类型集合，数组形式
   */
  shareMode?: Array<'qq' | 'qzone' | 'quickToDialog' | 'wechatFriends' | 'wechatMoment' | 'recentContacts'>;
  /**
   * QQ 小程序 1.17.0
   * 发送对象的 FriendInfo
   */
  shareMessageFriendInfo?: any;
  /**
   * QQ 小程序 1.17.0
   * 转发标题，不传则默认使用当前小程序的昵称。
   */
  shareMessageTitle?: string;
  /**
   * QQ 小程序 1.17.0
   * 转发显示图片的链接，可以是网络图片路径（仅 QQ CDN 域名路径）或本地图片文件路径或相对代码包根目录的图片文件路径。显示图片长宽比是 5:4
   */
  shareMessageImg?: string;
  /**
   * 1.5.0
   * 是否显示会话内消息卡片，设置此参数为 true，用户进入客服会话会在右下角显示"可能要发送的小程序"提示，
   * 用户点击后可以快速发送小程序消息，open-type="contact"时有效
   */
  showMessageCard?: boolean;
  /**
   * 1.3.0
   * 用户点击该按钮时，会返回获取到的用户信息，回调的detail数据与wx.getUserInfo返回的一致，
   * open-type="getUserInfo"时有效
   */
  onGetUserInfo?: (event: GenericEvent) => any;
  /**
   * 1.5.0
   * 客服消息回调，open-type="contact"时有效
   */
  onContact?: (event: GenericEvent) => any;
  /**
   * 1.2.0
   * 获取用户手机号回调，open-type=getPhoneNumber时有效
   */
  onGetPhoneNumber?: (event: GenericEvent) => any;
  /**
   * 2.21.2
   * 获取用户手机号回调，open-type=getPhoneNumber时有效
   */
  onChooseAvatar?: (event: GenericEvent) => any;
  /**
   * 1.9.5
   * 当使用开放能力时，发生错误的回调，open-type=launchApp时有效
   */
  onError?: (event: GenericEvent) => any;
  /**
   * 2.0.7
   * 在打开授权设置页后回调，open-type=openSetting时有效
   */
  onOpenSetting?: (event: GenericEvent) => any;
  /**
   * 2.4.4
   * 打开 APP 成功的回调，open-type=launchApp时有效
   */
  onLaunchApp?: (event: GenericEvent) => any;
  /**
   * QQ 小程序 1.0.0
   * 添加好友的回调
   */
  onAddFriend?: (event: GenericEvent) => any;
  /**
   * QQ 小程序 1.16.0
   * 添加群应用的回调
   */
  onAddGroupApp?: (event: GenericEvent) => any;
}

/**
 * https://developers.weixin.qq.com/miniprogram/dev/component/button.html
 * https://q.qq.com/wiki/develop/miniprogram/component/form/button.html
 */
export const Button: React.ComponentType<ButtonProps> = createHostComponent<ButtonProps>('button');

Button.defaultProps = {
  hoverClassName: 'button-hover',
  hoverStartTime: 20,
  hoverStayTime: 70,
};
