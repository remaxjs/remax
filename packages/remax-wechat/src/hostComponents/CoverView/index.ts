import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps } from '../../types/component';

export interface CoverViewProps extends BaseProps {
  /**	设置顶部滚动偏移量，仅在设置了 overflow-y: scroll 成为滚动元素后生效	2.1.0  */
  scrollTop?: number | string;
  markerId?: string;
  slot?: string;
}

/**
 * @see https://developers.weixin.qq.com/miniprogram/dev/component/cover-view.html
 */
export const CoverView: React.ComponentType<CoverViewProps> = createHostComponent<CoverViewProps>('cover-view');
