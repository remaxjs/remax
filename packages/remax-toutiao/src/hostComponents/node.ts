import * as button from './Button/node';
import * as canvas from './Canvas/node';
import * as checkboxGroup from './CheckboxGroup/node';
import * as checkbox from './Checkbox/node';
import * as form from './Form/node';
import * as icon from './Icon/node';
import * as image from './Image/node';
import * as input from './Input/node';
import * as label from './Label/node';
import * as navigator from './Navigator/node';
import * as pickerViewColumn from './PickerViewColumn/node';
import * as pickerView from './PickerView/node';
import * as picker from './Picker/node';
import * as progress from './Progress/node';
import * as radioGroup from './RadioGroup/node';
import * as radio from './Radio/node';
import * as richText from './RichText/node';
import * as scrollView from './ScrollView/node';
import * as slider from './Slider/node';
import * as swiperItem from './SwiperItem/node';
import * as swiper from './Swiper/node';
import * as switchComponent from './Switch/node';
import * as text from './Text/node';
import * as textarea from './Textarea/node';
import * as video from './Video/node';
import * as view from './View/node';
import * as webView from './WebView/node';
import * as movableArea from './MovableArea/node';
import * as movableView from './MovableView/node';

const hostComponents = new Map();

hostComponents.set('button', button);
hostComponents.set('canvas', canvas);
hostComponents.set('checkbox-group', checkboxGroup);
hostComponents.set('checkbox', checkbox);
hostComponents.set('form', form);
hostComponents.set('icon', icon);
hostComponents.set('image', image);
hostComponents.set('input', input);
hostComponents.set('label', label);
hostComponents.set('navigator', navigator);
hostComponents.set('picker-view-column', pickerViewColumn);
hostComponents.set('picker-view', pickerView);
hostComponents.set('picker', picker);
hostComponents.set('progress', progress);
hostComponents.set('radio-group', radioGroup);
hostComponents.set('radio', radio);
hostComponents.set('rich-text', richText);
hostComponents.set('scroll-view', scrollView);
hostComponents.set('slider', slider);
hostComponents.set('swiper-item', swiperItem);
hostComponents.set('swiper', swiper);
hostComponents.set('switch', switchComponent);
hostComponents.set('text', text);
hostComponents.set('textarea', textarea);
hostComponents.set('video', video);
hostComponents.set('view', view);
hostComponents.set('web-view', webView);
hostComponents.set('movable-area', movableArea);
hostComponents.set('movable-view', movableView);

export default hostComponents;
