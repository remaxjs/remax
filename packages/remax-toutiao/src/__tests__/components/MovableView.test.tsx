import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import { MovableView } from '../../hostComponents';

describe('MovableView', () => {
  it('render correctly', () => {
    const testRenderer = TestRenderer.create(<MovableView />);

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
