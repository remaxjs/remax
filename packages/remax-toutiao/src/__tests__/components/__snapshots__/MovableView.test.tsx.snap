// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovableView render correctly 1`] = `
<movable-view
  animation={true}
  damping={20}
  direction="none"
  disabled={false}
  friction={2}
  height={10}
  inertia={false}
  outOfBounds={false}
  scale={false}
  scaleMax={10}
  scaleMin={0}
  scaleValue={1}
  width={10}
/>
`;
